compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -3595,19 +3595,79 @@ object Types {
           case _ =>
             NoType
         }
-        if (defn.isCompiletime_S(tycon.symbol) && args.length == 1)
-          trace(i"normalize S $this", typr, show = true) {
-            args.head.normalized match {
-              case ConstantType(Constant(n: Int)) if n >= 0 && n < Int.MaxValue =>
-                ConstantType(Constant(n + 1))
-              case none => tryMatchAlias
-            }
-          }
-        else tryMatchAlias
+
+        tryCompiletimeConstantFold.getOrElse(tryMatchAlias)
+
       case _ =>
         NoType
     }
 
+    def tryCompiletimeConstantFold(implicit ctx: Context): Option[Type] = tycon match {
```

Comment:
No need to return an `Option` here, you can just return a `Type` and use `NoType` instead of `None`. (Types have an `orElse` method that would replace the `getOrElse` in `tryNormalize`)

Reply:
:+1: Done in 5e287cda8b18dd9c90bff69eb782e23afb820af9